py/desispec/scripts/link_calibnight.py

```diff
@@ -0,0 +1,138 @@
+
+import os, sys
+import argparse
+
+from desiutil.log import get_logger
+
+from desispec.io import findfile
+from desispec.io.util import decode_camword, relsymlink
+
+def parse(options=None):
+    """parse options from sys.argv or input list of options"""
+    p = argparse.ArgumentParser()
+    p.add_argument('--refnight', type=int, required=True,
+                   help='Reference night with calibs')
+    p.add_argument('--newnight', type=int, required=True,
+                   help='New night without calibs, to link to refnight')
+    p.add_argument('-c', '--cameras', type=str, default='a0123456789',
+                   help='Camword of cameras to link, e.g. a0123b6r7z8')
+    p.add_argument('--include', type=str, 
+                   default='badcolumns,biasnight,fiberflatnight,psfnight,ctecorr',
+                   help='prefixes of types of calibnight files to create links')
+    p.add_argument('--exclude', type=str, 
+                   help='prefixes of types of calibnight files to exclude from links')
+    p.add_argument('--dryrun', action='store_true',
+                   help="dry run; don't actually create links")
+
+    args = p.parse_args(options)
+
+    if args.exclude is None:
+        args.exclude = set()
+    else:
+        args.exclude = set(args.exclude.split(','))
+
+    args.include = set(args.include.split(',')) - args.exclude
```

Comment:
I'd personally prefer if the default of include was None, and here you added logic:
1) If `include` is defined by the user, make sure that the set is independent of the `exclude` set. Then tell the user that only one or the other is used and you will proceed with the `include` list.
2) `If include is None`, then define it as the `set('badcolumns,biasnight,fiberflatnight,psfnight,ctecorr'.split(','))` - set().

Reply:
Done, with modifications: I made --include and --exclude mutually exclusive rather than warning and proceeding with using only include.  I also check that --exclude is a strict subset of the default include so that the user doesn't accidentally not exclude something due to a typo.